lost/src/main/java/com/mapzen/android/lost/internal/FusedLocationProviderServiceImpl.java

```diff
@@ -0,0 +1,137 @@
+package com.mapzen.android.lost.internal;
+
+import com.mapzen.android.lost.api.LocationListener;
+import com.mapzen.android.lost.api.LocationRequest;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.util.Log;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.mapzen.android.lost.api.FusedLocationProviderApi.KEY_LOCATION_CHANGED;
+
+public class FusedLocationProviderServiceImpl implements LocationEngine.Callback {
+
+  private static final String TAG = FusedLocationProviderServiceImpl.class.getSimpleName();
+
+  private Context context;
+
+  private boolean mockMode;
+  private LocationEngine locationEngine;
+  private Map<LocationListener, LocationRequest> listenerToRequest;
+  private Map<PendingIntent, LocationRequest> intentToRequest;
+
+  public FusedLocationProviderServiceImpl(Context context) {
+    this.context = context;
+    locationEngine = new FusionEngine(context, this);
+    listenerToRequest = new HashMap<>();
+    intentToRequest = new HashMap<>();
+  }
+
+  public void shutdown() {
+    listenerToRequest.clear();
```

Comment:
I missed this when adding support for `PendingIntent` updates but `shutdown()` should also clear `intentToRequest`.